app/policies/division_policy.rb

```diff
@@ -18,4 +18,30 @@ def update?
   def destroy?
     division_admin(division: @record)
   end
+
+  def select?
+    show?
+  end
+
+  class Scope < Scope
+    def resolve
+      scope.where(id: accessible_ids)
+    end
+
+    # This merges in child divisions of the divisions for which a user has been specifically
+    # granted access.
+    def accessible_ids
+      all_ids = base_accessible_ids.map do |id|
+        division = Division.find_safe(id)
+        division.self_and_descendants.pluck(:id) if division
+      end
+      all_ids.flatten.uniq.compact
```

Comment:
I think you need a `compact` after `flatten` to remove any `nil`s in case divisions aren't found.